### ISSUE TITLE
EBL absorption affecting probabilities at low energies

### ISSUE THREAD

**Issue by specktakel** [me-manu/gammaALPs#12]
Inserting EBL between some ICM field and the GMF changes the photon survival probability below the energy at which EBL absorption becomes relevant.
This shows only when there actually is an environment after the EBL, so ICM + EBL is fine. I changed your source code in gammaALPs.core in the method run() to treat EBL absorption as a transfer matrix:
```
if OptDepth in [type(t) for t in self.modules]:
            # get the index of the EBL module
            idx = [type(t) for t in self.modules].index(OptDepth)            # convert self._atten to transfer matrices
            t_ebl = np.zeros((self.EGeV.size, 3, 3))
            t_ebl[:, 0, 0] = np.sqrt(self._atten)
            t_ebl[:, 1, 1] = np.sqrt(self._atten)
            t_ebl[:, 2, 2] = 1
            t = [t_ebl]
            t_ebl = np.array(t)
            #print(idx, t_ebl)            #insert in list of self._Tenv
            self._Tenv.insert(idx, t_ebl)
```
which is then multiplied with all the other transfer matrices. With this modification, I get consistent results.
Below is a plot showing this, ALP mass is 2neV, coupling constant is above each panel, GMF model is pshirkov with BSS. "struc" is a magnetic field model I implemented via MixFromArray, https://arxiv.org/abs/1908.03084 shows the field in Figure 1. Source is NGC 1275 with all other parameters from the example notebook. In my plot, the left side shows the "old" version of core.py, the right uses my modified version. The difference is on the left side between the green and red plot, they should be the same up to some 200 GeV, but they are obviously not.

![gammaalps_struc_ebl_gmf_comparison](https://user-images.githubusercontent.com/72220660/116361772-9e590580-a801-11eb-83f9-e5657df244cb.png)

**Comment by me-manu** [comment]
Thank you @specktakel for checking this and suggesting a solution. I vaguely remember that I did something similar years ago, but I don't recall why I opted for the currently implemented solution which assumes a "measurement" of the polarization states after the photon-ALP beam leaves the source. This is of course not realized in real life but seemed like an okay assumption back then. But your changes appear to work and to make sense to me. Can you make a pull request with your new code so that all tests are automatically run and I can review the code changes better?

**Comment by me-manu** [comment]
This has been fixed with PR #15
